src/pairings.rs

```diff
@@ -6,6 +6,9 @@ use core::ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub, SubAssign};
 
 use subtle::{Choice, ConditionallySelectable, ConstantTimeEq};
 
+#[cfg(feature = "pairings")]
+use alloc::vec::Vec;
```

Comment:
The `pairings` feature flag is only necessary for this more efficient pairing API; pairings can still be computed without `alloc`. The documentation (and possibly the feature flag name) should be updated to reflect this. Non-blocking, can be done in a subsequent PR.